src/components/App.js

```diff
@@ -1,20 +1,43 @@
-import React from 'react';
+import React, { Component } from 'react';
 import ButtonPanel from './ButtonPanel';
 import Display from './Display';
 import calculate from '../logic/calculate';
 
-const App = () => {
-  const dataObject = {
-    total: 45,
-    next: 56,
-    operation: 'X',
-  };
-  const buttonName = '-';
-  return (
-    <div className="calculator">
-      <Display />
-      <ButtonPanel calculate={calculate(dataObject, buttonName)} />
-    </div>
-  );
-};
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      total: '',
+      next: '',
+      operation: '',
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
```

Comment:
For proper initialization, pass `props` to the class constructor and the `super` call. i.e.
```javascript
constructor(props) {
  super(props);
...
```